onnx/test/parser_test.py

```diff
@@ -116,5 +116,74 @@ def test_parse_function_with_attributes(self) -> None:
         checker.check_model(model)
 
 
+    def test_composite_parse_function_with_attributes(self) -> None:
+        function_text = """
+         <
+         domain: "custom_domain",
+         opset_import: [ "" : 15],
+         doc_string: "Test function proto"
+         >
+           Selu
+           <alpha: float=1.67326319217681884765625, gamma: float=1.05070102214813232421875>
+           (X) => (C)
+           {
+               constant_alpha = Constant<value_float: float=@alpha>()
+               constant_gamma = Constant<value_float: float=@gamma>()
+               alpha_x = CastLike(constant_alpha, X)
+               gamma_x = CastLike(constant_gamma, X)
+               exp_x = Exp(X)
+               alpha_x_exp_x = Mul(alpha_x, exp_x)
+               alpha_x_exp_x_ = Sub(alpha_x_exp_x, alpha_x)
+               neg = Mul(gamma_x, alpha_x_exp_x_)
+               pos = Mul(gamma_x, X)
+               _zero = Constant<value_float=0.0>()
+               zero = CastLike(_zero, X)
+               less_eq = LessOrEqual(X, zero)
+               C = Where(less_eq, neg, pos)
+           }
+        """
+
+        functions = [onnx.parser.parse_function(function_text)]
```

Comment:
May be useful to check/assert that the returned FunctionProto has the expected attributes.

Reply:
At least check `len(f.attribute_protos)`.

Reply:
added verification for parser outcomes.

Reply:
I don't see it. May not be checked in yet?